components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useState } from 'react';
+import { ArrowLeft, ArrowRight } from 'react-feather';
+
+import {
+  PaginationButtonGroupProps,
+  PaginationButtonProps,
+  PaginationNumberButtonProps,
+  PaginationProps
+} from '@/components/Pagination/Pagination.types';
+import { calculatePagination } from '@/utils/pagination.util';
+
+const PrevButton = (props: PaginationButtonProps) => {
+  return (
+    <div className="flex flex-1 justify-start">
+      {props.show && (
+        <button
+          className="flex gap-2 rounded-lg py-2 px-4 text-gray-500 hover:bg-gray-100"
+          onClick={() => props.onPageClick()}
+        >
+          <ArrowLeft></ArrowLeft> Önceki
+        </button>
+      )}
+    </div>
+  );
+};
+
+const NextButton = (props: PaginationButtonProps) => {
+  return (
+    <div className="flex flex-1 justify-end">
+      {props.show && (
+        <button
+          className="flex gap-2 rounded-lg py-2 px-4 text-gray-500 hover:bg-gray-100"
+          onClick={() => props.onPageClick()}
+        >
+          Sonraki <ArrowRight></ArrowRight>
+        </button>
+      )}
+    </div>
+  );
+};
+
+const PaginationNumberButton = (props: PaginationNumberButtonProps) => {
+  return (
+    <button
+      className={`rounded-lg py-2 px-4 hover:bg-pink-100 hover:text-pink-400 ${
+        props.active ? 'bg-pink-100 text-pink-600' : 'text-gray-500'
+      }`}
+      onClick={() => (props.onPageClick ? props.onPageClick() : null)}
+    >
+      {props.page}
+    </button>
+  );
+};
+
+const PaginationButtonGroup = (props: PaginationButtonGroupProps) => {
+  const [totalCount, setTotalCount] = useState(props.pages.length);
+  const [centerIndex, setCenterIndex] = useState(0);
+
+  useEffect(() => {
+    const index = props.pages.findIndex(p => p === props.current);
+    setCenterIndex(index);
+    setTotalCount(props.pages.length);
+  }, [props.pages, props.current]);
+
+  return (
+    <div className="flex flex-1 grow justify-center gap-2">
+      {centerIndex > 1 && (
+        <>
+          <PaginationNumberButton
```

Comment:
We should not use same component repeatedly. You should manipulate pagination data.